src/Sarif.Multitool/PageCommand.cs

```diff
@@ -0,0 +1,195 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis.Sarif.Map;
+using Newtonsoft.Json;
+
+namespace Microsoft.CodeAnalysis.Sarif.Multitool
+{
+    internal class PageCommand : CommandBase
+    {
+        private readonly IFileSystem _fileSystem;
+
+        public PageCommand(IFileSystem fileSystem = null)
+        {
+            _fileSystem = fileSystem ?? new FileSystem();
+        }
+
+        public int Run(PageOptions options)
+        {
+            try
+            {
+                RunWithoutCatch(options);
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex);
+                return 1;
+            }
+
+            return 0;
+        }
+
+        public void RunWithoutCatch(PageOptions options)
+        {
+            if (options.RunIndex < 0) { throw new ArgumentOutOfRangeException("runIndex"); }
+            if (options.Index < 0) { throw new ArgumentOutOfRangeException("index"); }
+            if (options.Count < 0) { throw new ArgumentOutOfRangeException("count"); }
+            if (!_fileSystem.FileExists(options.InputFilePath)) { throw new FileNotFoundException($"Input file \"{options.InputFilePath}\" not found."); }
+
+            if (options.Force == false && _fileSystem.FileExists(options.OutputFilePath))
+            {
+                Console.WriteLine($"Output file \"{options.OutputFilePath}\" already exists. Stopping.");
+                return;
+            }
+
+            // Load the JsonMap, if previously built and up-to-date, or rebuild it
+            string mapPath = Path.ChangeExtension(options.InputFilePath, ".map.json");
```

Comment:
>string mapPath = Path.ChangeExtension(options.InputFilePath, ".map.json"); [](start = 12, length = 74)

NIT: we could move this processing inside the method since its an internal information.. 


unless... the idea here was to ensure we reusability of the method if someone chooses a different path for storing the map file, in which case, i relent :)

just curious!

Reply:
Yep, good suggestion. This was another facet of a prior factoring.